code/modules/mob/living/carbon/carbon.dm

```diff
@@ -451,26 +452,19 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 			if(!C.check_clothing(src))//return values confuse me right now
 				return
 
-	var/obj/machinery/atmospherics/unary/vent_found
+	var/obj/machinery/atmospherics/entrance_found = clicked_on
 
-	if(clicked_on)
-		vent_found = clicked_on
-		if(!istype(vent_found) || !vent_found.can_crawl_through())
-			vent_found = null
+	if(!(is_type_in_list(entrance_found, GLOB.ventcrawl_machinery)) && !(istype(entrance_found, /obj/machinery/atmospherics) && entrance_found.check_open()))
+		entrance_found = null
 
+	if(!entrance_found)
+		to_chat(src, "<span class='warning'>This ventilation duct is not connected to anything!</span>")
+		return
 
-	if(!vent_found)
-		for(var/obj/machinery/atmospherics/machine in range(1,src))
-			if(is_type_in_list(machine, GLOB.ventcrawl_machinery) && machine.can_crawl_through())
-				vent_found = machine
-				break
-
-	if(vent_found)
-		if(vent_found.parent && (vent_found.parent.members.len || vent_found.parent.other_atmosmch))
-			visible_message("<span class='notice'>[src] begins climbing into the ventilation system...</span>", \
-							"<span class='notice'>You begin climbing into the ventilation system...</span>")
-
-			if(!do_after(src, 45, target = src))
+	if(entrance_found)
+		var/datum/pipeline/P = entrance_found.returnPipenet()
+		if(P && (length(P.members) || P.other_atmosmch))
```

Comment:
Doesnt this make it so you cant climb into single device pipenets? If so you need to document this change

Reply:
I went back and checked, you are still able to climb into single device pipe nets.